Streetcode/Streetcode.BLL/MediatR/Authentication/Register/RegisterHandler.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Text.RegularExpressions;
+using AutoMapper;
+using FluentResults;
+using MediatR;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Streetcode.BLL.DTO.Authentication.Register;
+using Streetcode.BLL.DTO.Users;
+using Streetcode.BLL.Interfaces.Logging;
+using Streetcode.DAL.Entities.Users;
+using Streetcode.DAL.Enums;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+
+namespace Streetcode.BLL.MediatR.Authentication.Register
+{
+    public class RegisterHandler : IRequestHandler<RegisterQuery, Result<RegisterResponseDTO>>
+    {
+        private readonly ILoggerService _logger;
+        private readonly IMapper _mapper;
+        private readonly IRepositoryWrapper _repositoryWrapper;
+        private readonly UserManager<User> _userManager;
+
+        public RegisterHandler(IRepositoryWrapper repositoryWrapper, ILoggerService logger, IMapper mapper, UserManager<User> userManager)
+        {
+            _repositoryWrapper = repositoryWrapper;
+            _logger = logger;
+            _mapper = mapper;
+            _userManager = userManager;
+        }
+
+        public async Task<Result<RegisterResponseDTO>> Handle(RegisterQuery request, CancellationToken cancellationToken)
+        {
+            User user = new User()
```

Comment:
You should delegate it to mapper, I think it has all needed functionality for that kind of map